codegen/smithy-aws-typescript-codegen/src/main/java/software/amazon/smithy/aws/typescript/codegen/RestXmlResponseProtocolGenerator.java

```diff
@@ -0,0 +1,137 @@
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.smithy.aws.typescript.codegen;
+
+import java.util.List;
+import java.util.Set;
+import software.amazon.smithy.aws.traits.ServiceTrait;
+import software.amazon.smithy.codegen.core.SymbolProvider;
+import software.amazon.smithy.codegen.core.SymbolReference;
+import software.amazon.smithy.model.knowledge.HttpBinding;
+import software.amazon.smithy.model.shapes.MemberShape;
+import software.amazon.smithy.model.shapes.Shape;
+import software.amazon.smithy.typescript.codegen.TypeScriptWriter;
+import software.amazon.smithy.typescript.codegen.integration.HttpBindingProtocolGenerator;
+
+/**
+ * Handles general components across the XML-document response based AWS protocols
+ * that have HTTP bindings. It handles reading from document bodies, including
+ * generating any functions needed for performing deserialization.
+ *
+ * This builds on the foundations of the {@link HttpBindingProtocolGenerator} to
+ * handle components of binding to HTTP responses.
+ *
+ * @see XmlShapeDeserVisitor
+ * @see XmlMemberDeserVisitor
+ * @see AwsProtocolUtils
+ * @see <a href="https://awslabs.github.io/smithy/spec/http.html">Smithy HTTP protocol bindings.</a>
+ * @see <a href="https://awslabs.github.io/smithy/spec/xml.html">Smithy XML traits.</a>
+ */
+abstract class RestXmlResponseProtocolGenerator extends HttpBindingProtocolGenerator {
+
+    RestXmlResponseProtocolGenerator() {
+        super(true);
+    }
+
+    @Override
+    protected void generateDocumentBodyShapeDeserializers(GenerationContext context, Set<Shape> shapes) {
+        AwsProtocolUtils.generateDocumentBodyShapeSerde(context, shapes, new XmlShapeDeserVisitor(context));
+    }
+
+    @Override
+    public void generateSharedComponents(GenerationContext context) {
+        super.generateSharedComponents(context);
+
+        TypeScriptWriter writer = context.getWriter();
+
+        // Include an XML body parser used to deserialize documents from HTTP responses.
+        writer.addImport("SerdeContext", "__SerdeContext", "@aws-sdk/types");
+        writer.addDependency(AwsDependency.XML_PARSER);
+        writer.addDependency(AwsDependency.XML_PARSER_TYPES);
+        writer.addImport("parse", "pixlParse", "pixl-xml");
+        writer.openBlock("const parseBody = (streamBody: any, context: __SerdeContext): any => {", "};", () -> {
+            writer.openBlock("return collectBodyString(streamBody, context).then(encoded => {", "});", () -> {
+                writer.openBlock("if (encoded.length) {", "}", () -> {
+                    writer.write("return pixlParse(encoded);");
```

Comment:
`preserveAttribute` option is required here. See: https://github.com/aws/aws-sdk-js-v3/blob/17e85a9a4021b4f26118853ae5a0c8d76f66e354/packages/xml-body-parser/src/index.ts#L35-L36

This option is default to `false`. See [here](https://github.com/jhuckaby/pixl-xml/blob/67a470dad7319c88b6ab086562d0ba48a866d722/xml.js#L78)

Reply:
Leaving this as `false` means the parser flattens these in the intermediate representation, slightly simplifying the code generation.